diff --git a/README.md b/README.md
index 713c5c7..150a025 100644
--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
-efivar
+# efivar
 ======
 
-Tools and libraries to manipulate EFI variables
+## Tools and libraries to manipulate EFI variables
 ---------------------------------------------
 
 This library is free software; you can redistribute it and/or
@@ -22,7 +22,26 @@ There is an ABI tracker for this project at [ABI Laboratory].
 [http://www.gnu.org/licenses/]: http://www.gnu.org/licenses/
 [ABI Laboratory]: https://abi-laboratory.pro/tracker/timeline/efivar/
 
-WARNING
+## Build
+
+### Environment
+
+```
+env | grep CFLAGS
+CFLAGS= -resource-dir=/usr/lib/clang/14.0.3 -B/usr/bin -Os -fPIC -mtune=native
+env | grep CXXFLAGS
+CXXFLAGS=-DLIBCXX_HAS_MUSL_LIBC=1 -D_LIBCPP_HAS_MUSL_LIBC=1 -D_LIBCPP_PROVIDES_DEFAULT_RUNE_TABLE=1  -resource-dir=/usr/lib/clang/14.0.3 -B/usr/bin -Os -fPIC -mtune=native
+env | grep LDFLAGS
+
+```
+
+### Command
+
+```
+CC=/usr/bin/clang CXX=/usr/bin/clang++ LD=/usr/bin/ld.bfd make -j13
+CC=/usr/bin/clang CXX=/usr/bin/clang++ LD=/usr/bin/ld.bfd doas make install
+```
+## WARNING
 =======
 You should probably not run "make a brick" *ever*, unless you're already
 reasonably sure it won't permanently corrupt your firmware.  This is not a
diff --git a/src/include/defaults.mk b/src/include/defaults.mk
index b8cc590..49bfb5a 100644
--- a/src/include/defaults.mk
+++ b/src/include/defaults.mk
@@ -21,14 +21,14 @@ $(call set-if-undefined,HOSTCC,$(COMPILER))
 $(call set-if-undefined,HOSTCCLD,$(HOSTCC))
 
 # temporary, see https://sourceware.org/bugzilla/show_bug.cgi?id=28264
-#OPTIMIZE_GCC = -flto
-OPTIMIZE_GCC =
+# OPTIMIZE_GCC = -flto
+# OPTIMIZE_GCC =
 OPTIMIZE ?= -Og $(call family,OPTIMIZE)
 DEBUGINFO ?= -g3
-WARNINGS_GCC ?=
-WARNINGS_CCC_ANALYZER ?= $(WARNINGS_GCC)
+# WARNINGS_GCC ?=
+# WARNINGS_CCC_ANALYZER ?= $(WARNINGS_GCC)
 WARNINGS ?= -Wall -Wextra $(call family,WARNINGS)
-ERRORS_GCC ?=
+# ERRORS_GCC ?=
 ERRORS ?= -Werror $(call family,ERRORS)
 CPPFLAGS ?=
 override _CPPFLAGS := $(CPPFLAGS)
@@ -36,8 +36,8 @@ override CPPFLAGS = $(_CPPFLAGS) -DLIBEFIVAR_VERSION=$(VERSION) \
 	    -D_GNU_SOURCE \
 	    -I$(TOPDIR)/src/include/
 CFLAGS ?= $(OPTIMIZE) $(DEBUGINFO) $(WARNINGS) $(ERRORS)
-CFLAGS_GCC ?= -specs=$(TOPDIR)/src/include/gcc.specs \
-	      -fno-merge-constants
+# CFLAGS_GCC ?= -specs=$(TOPDIR)/src/include/gcc.specs \
+# 	      -fno-merge-constants
 override _CFLAGS := $(CFLAGS)
 override CFLAGS = $(_CFLAGS) \
 		  -std=gnu11 \
@@ -51,7 +51,7 @@ LDFLAGS ?=
 override _CCLDFLAGS := $(CCLDFLAGS)
 override _LDFLAGS := $(LDFLAGS)
 override LDFLAGS = $(CFLAGS) -L. $(_LDFLAGS) $(_CCLDFLAGS) \
-		   -Wl,--add-needed \
+           -fuse-ld=bfd \
 		   -Wl,--build-id \
 		   -Wl,--no-allow-shlib-undefined \
 		   -Wl,--no-undefined-version \
@@ -62,7 +62,7 @@ override LDFLAGS = $(CFLAGS) -L. $(_LDFLAGS) $(_CCLDFLAGS) \
 		   $(call family,LDFLAGS) $(call family,CCLDFLAGS) \
 		   $(call pkg-config-ccldflags)
 override CCLDFLAGS = $(LDFLAGS)
-SOFLAGS_GCC =
+# SOFLAGS_GCC =
 SOFLAGS_CLANG =
 SOFLAGS ?=
 override _SOFLAGS := $(SOFLAGS)
@@ -85,20 +85,20 @@ HOST_CPPFLAGS ?= $(CPPFLAGS)
 override _HOST_CPPFLAGS := $(HOST_CPPFLAGS)
 override HOST_CPPFLAGS = $(_HOST_CPPFLAGS) \
 			 -DEFIVAR_BUILD_ENVIRONMENT $(HOST_MARCH)
-HOST_CFLAGS_GCC ?=
+# HOST_CFLAGS_GCC ?=
 HOST_CFLAGS_CLANG ?=
 HOST_CFLAGS ?= $(CFLAGS) $(call family,HOST_CFLAGS)
 override _HOST_CFLAGS := $(HOST_CFLAGS)
 override HOST_CFLAGS = $(_HOST_CFLAGS)
 HOST_LDFLAGS_CLANG ?= -Wl,--fatal-warnings,-z,relro -rtlib=compiler-rt
-HOST_LDFLAGS_GCC ?= -Wl,--no-undefined-version
+# HOST_LDFLAGS_GCC ?= -Wl,--no-undefined-version
 HOST_LDFLAGS ?=
 HOST_CCLDFLAGS ?=
 override _HOST_LDFLAGS := $(HOST_LDFLAGS)
 override _HOST_CCLDFLAGS := $(HOST_CCLDFLAGS)
 override HOST_LDFLAGS = $(HOST_CFLAGS) -L. \
 			$(_HOST_LDFLAGS) $(_HOST_CCLDFLAGS) \
-			-Wl,--add-needed \
+            -fuse-ld=bfd \
 			-Wl,--build-id \
 			-Wl,--no-allow-shlib-undefined \
 			-Wl,-z,now \
@@ -110,6 +110,7 @@ override HOST_CCLDFLAGS = $(HOST_LDFLAGS)
 
 PKG_CONFIG = $(shell if [ -e "$$(env $(CROSS_COMPILE)pkg-config 2>&1)" ]; then echo $(CROSS_COMPILE)pkg-config ; else echo pkg-config ; fi)
 INSTALL ?= install
+LD  := $(CROSS_COMPILE)ld.bfd
 AR	:= $(CROSS_COMPILE)$(COMPILER)-ar
 NM	:= $(CROSS_COMPILE)$(COMPILER)-nm
 RANLIB	:= $(CROSS_COMPILE)$(COMPILER)-ranlib
diff --git a/src/include/gcc.specs b/src/include/gcc.specs
index ef28e2b..d85e865 100644
--- a/src/include/gcc.specs
+++ b/src/include/gcc.specs
@@ -5,4 +5,4 @@
 + %{!shared:%{!static:%{!r:-pie}}} %{static:-Wl,-no-fatal-warnings -Wl,-static -static -Wl,-z,relro,-z,now} -grecord-gcc-switches
 
 *link:
-+ %{!static:--fatal-warnings} --no-undefined-version --no-allow-shlib-undefined --add-needed -z now --build-id %{!static:%{!shared:-pie}} %{shared:-z relro} %{static:%<pie}
++ %{!static:--fatal-warnings} --no-undefined-version --no-allow-shlib-undefined -z now --build-id %{!static:%{!shared:-pie}} %{shared:-z relro} %{static:%<pie}
